### PR TITLE
Validate datadog annotations

### PR DESCRIPTION
**Pull request by adammw**
Validate datadog annotations, if present. Mistyping an annotation with the incorrect container name can leave you wondering why it's not working for an hour. Fail fast.

@zendesk/compute @cholman-zd 

### Risks
- Low: breaks deployment of weird datadog-annotated deploys

### CODE REVIEW

File: plugins/kubernetes/app/models/kubernetes/role_validator.rb [hunk 1]
```diff
@@ -178,6 +181,22 @@ def validate_api_version
...
     end
 
+    # validate datadog-specific annotations against
+    # https://docs.datadoghq.com/agent/autodiscovery/integrations/?tab=kubernetes#configuration
+    def validate_datadog_annotations
+      templates.each do |template|
+        pod_spec = pod_template(template)
+        annotations = pod_spec.dig(:metadata, :annotations) || {}
+        containers = pod_spec.dig(:spec, :containers) || []
+        dd_container_names = annotations.keys.map { |k| (DATADOG_AD_REGEXP.match(k) || [])[1] }.compact.uniq
```
Comment (grosser) [inline plugins/kubernetes/app/models/kubernetes/role_validator.rb]:
```suggestion
        dd_container_names = annotations.keys.map { |k| k[DATADOG_AD_REGEXP, 1] }.compact.uniq
```

File: plugins/kubernetes/app/models/kubernetes/role_validator.rb [hunk 2]
```diff
@@ -178,6 +181,22 @@ def validate_api_version
...
+        spec_container_names = containers.map { |c| c[:name] }.compact
+
+        unless (invalid = dd_container_names - spec_container_names).empty?
```
Comment (grosser) [inline plugins/kubernetes/app/models/kubernetes/role_validator.rb]: could also do this ... but about the same kind of funky 🤷‍♂ 
```suggestion
        unless invalid = (dd_container_names - spec_container_names).presence
```

File: plugins/kubernetes/app/models/kubernetes/role_validator.rb [hunk 3]
```diff
@@ -398,6 +417,17 @@ def validate_host_volume_paths
 
     # helpers below
 
+    def pod_template(template)
```
Comment (grosser) [inline plugins/kubernetes/app/models/kubernetes/role_validator.rb]: that's what `def templates` already does

**Review by grosser — COMMENTED** [review]
I like the idea, annoying error to debug  :)

File: plugins/kubernetes/app/models/kubernetes/role_validator.rb (hunk 3 above)
Comment (adammw) [reply plugins/kubernetes/app/models/kubernetes/role_validator.rb]: i don't understand how...

Comment (grosser) [reply plugins/kubernetes/app/models/kubernetes/role_validator.rb]: delegates to role_config_file.rb self.templates ... which does some magic to do it ... there are tests, so pretty sure it works :)

**Review by uthark — APPROVED** [review]

File: plugins/kubernetes/app/models/kubernetes/role_validator.rb (hunk 3 above)
Comment (adammw) [reply plugins/kubernetes/app/models/kubernetes/role_validator.rb]: updated.

**Pull request merged**